### PR TITLE
 Move Stored(File|Directory)Helper into NewPipe 

### PR DESCRIPTION
**Pull request by wb9688**
#### What is it?
- [ ] Bug fix (user facing)
- [ ] Feature (user facing)
- [x] Code base improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
This moves `StoredFileHelper` and `StoredDirectoryHelper` into NewPipe, while fixing a lot of code style issues.

#### Relies on the following changes
#3777

#### Agreement
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/streams/io/StoredDirectoryHelper.java [hunk 1]
```diff
@@ -52,47 +52,45 @@ public StoredDirectoryHelper(@NonNull Context context, @NonNull Uri path, String
...
 
-            String[] projection = new String[]{COLUMN_DISPLAY_NAME};
-            String selection = "(LOWER(" + COLUMN_DISPLAY_NAME + ") LIKE ?%";
-            ContentResolver cr = context.getContentResolver();
+            final String[] projection = new String[]{COLUMN_DISPLAY_NAME};
+            final String selection = "(LOWER(" + COLUMN_DISPLAY_NAME + ") LIKE ?%";
+            final ContentResolver cr = context.getContentResolver();
 
-            try (Cursor cursor = cr.query(docTreeChildren, projection, selection, new String[]{lcFilename}, null)) {
+            try (Cursor cursor = cr.query(docTreeChildren, projection, selection,
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/streams/io/StoredDirectoryHelper.java:88]: Can `final` be put here?
```suggestion
            try (final Cursor cursor = cr.query(docTreeChildren, projection, selection,
```

File: app/src/main/java/org/schabi/newpipe/streams/io/StoredDirectoryHelper.java [hunk 2]
```diff
@@ -218,72 +228,82 @@ public String toString() {
...
     //      Utils
     ///////////////////
 
-    private static void addIfStartWith(ArrayList<String> list, @NonNull String base, String str) {
-        if (str == null || str.isEmpty()) return;
-        str = str.toLowerCase();
-        if (str.startsWith(base)) list.add(str);
+    private static void addIfStartWith(final ArrayList<String> list, @NonNull final String base,
+                                       final String str) {
+        if (str == null || str.isEmpty()) {
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/streams/io/StoredDirectoryHelper.java:237]:
```suggestion
        if (isNullOrEmpty(str)) {
```

File: app/src/main/java/org/schabi/newpipe/streams/io/StoredDirectoryHelper.java [hunk 3]
```diff
@@ -218,72 +228,82 @@ public String toString() {
...
+                DocumentsContract.getDocumentId(tree.getUri()));
+        final String[] projection = {COLUMN_DISPLAY_NAME, COLUMN_DOCUMENT_ID};
+        final ContentResolver contentResolver = context.getContentResolver();
 
-        filename = filename.toLowerCase();
+        final String lowerFilename = filename.toLowerCase();
 
-        try (Cursor cursor = contentResolver.query(childrenUri, projection, selection, new String[]{filename}, null)) {
-            if (cursor == null) return null;
+        try (Cursor cursor = contentResolver.query(childrenUri, projection, selection,
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/streams/io/StoredDirectoryHelper.java:292]: `final`

File: app/src/main/java/org/schabi/newpipe/streams/io/StoredDirectoryHelper.java (hunk 1 above)
Comment (wb9688) [reply app/src/main/java/org/schabi/newpipe/streams/io/StoredDirectoryHelper.java]: No, it's `final` by default, and thus Checkstyle would complain about it being redundant

**Pull request closed without merging**